<?xml version="1.0" encoding="UTF-8"?>
<robot name="puzzlebot">
  <material name="white">
	<color rgba="1 1 1 1"/>
  </material>
  <material name="orange">
	<color rgba="1 0.3 0.1 1"/>
  </material>
  <material name="blue">
	<color rgba="0.2 0.2 1 1"/>
  </material>
  <material name="black">
	<color rgba="0 0 0 1"/>
  </material>

  <!-- ******   Base-Link   ****** -->
  <link name="base_link">
  </link>
  <!-- ******  CHASSIS  ****** -->
  <link name="chassis_link">
	<visual>
    	<origin xyz="0.05 0 0" rpy="${pi/2} 0 ${pi/2}"/>
    	<geometry>
        	<!--box size="0.3 0.3 0.15"/-->
            <mesh filename="package://bloque/meshes/MCR2_1000_13_Chassis.stl"/>
    	</geometry>
    	<material name="white"/>
	</visual>
  </link>


  <joint name="chassis_joint" type="fixed">
	<parent link="base_link"/>
	<child link="chassis_link"/>
	<origin xyz="-0.1 0 0"/>
  </joint>


  <!-- ******  LEFT WHEEL  ****** -->
  <joint name="left_wheel_joint" type="continuous">
	<parent link="base_link"/>
	<child link="left_wheel"/>
	<origin xyz="0 0.095 0" rpy="${pi/2} 0 0"/>
	<axis xyz="0 0 -1"/>
  </joint>
  <link name="left_wheel">
	<visual>
  	<geometry>
    	<!--cylinder length="0.04" radius="0.05" /-->
	<mesh filename="package://bloque/meshes/MCR2_1000_1_1_Wheel_Coupler_2.stl"/>
  	</geometry>
  	<material name="blue"/>
	</visual>
  </link>


 <!-- ******  RIGHT WHEEL  ****** -->
  <joint name="right_wheel_joint" type="continuous">
	<parent link="base_link"/>
	<child link="right_wheel"/>
	<origin xyz="0 -0.095 0" rpy="-${pi/2} 0 0"/>
	<axis xyz="0 0 1"/>
  </joint>
  <link name="right_wheel">
	<visual>
  	<geometry>
	<mesh filename="package://bloque/meshes/MCR2_1000_1_1_Wheel_Coupler_2.stl"/>
  	</geometry>
  	<material name="blue"/>
	</visual>
  </link>


</robot>
